servlet-core/src/main/java/io/micronaut/servlet/http/ServletBodyBinder.java

```diff
@@ -137,49 +117,66 @@ public List<ConversionError> getConversionErrors() {
                     Optional<T> result = conversionService.convert(servletHttpRequest.getParameters().asMap(), context);
                     return () -> result;
                 }
-            }
-            final MediaTypeCodec codec = mediaTypeCodeRegistry
-                    .findCodec(mediaType, type)
+            } else {
+                MessageBodyReader messageBodyReader = source.getAttribute(HttpAttributes.ROUTE_INFO, RouteInfo.class)
+                    .map(RouteInfo::getMessageBodyReader)
                     .orElse(null);
-
-            if (codec != null) {
-                try (InputStream inputStream = servletHttpRequest.getInputStream()) {
-                    if (Publishers.isConvertibleToPublisher(type)) {
-                        final Argument<?> typeArg = argument.getFirstTypeVariable().orElse(Argument.OBJECT_ARGUMENT);
-                        if (Publishers.isSingle(type)) {
-                            T content = (T) codec.decode(typeArg, inputStream);
-                            final Publisher<T> publisher = Publishers.just(content);
-                            final T converted = conversionService.convertRequired(publisher, type);
-                            return () -> Optional.of(converted);
+                if (name == null && messageBodyReader != null && messageBodyReader.isReadable(context.getArgument(), mediaType)) {
```

Comment:
i dont like how this logic is duplicated in core and here but i think it will have to wait until i come up with a response byte body api